### PR TITLE
Handle invalid value prop in DateTimePicker

### LINKED ISSUE #2068: Conditionally handle undefined value being passed to DateTimePicker
https://yedhin.neetorecord.com/watch/6d94a159-6f6b-40c1-ab17-7fbcdd519bb5

### PR DESCRIPTION
**Pull request by farhanlatheef**
- Fixes #2068 

**Description**
If the value prop is invalid date or date string, the value is converted to null.

**Checklist**

~- [ ] I have made corresponding changes to the documentation.~
- [x] I have updated the types definition of modified exports.
- [x] I have verified the functionality in some of the neeto web-apps.
- [x] I have added tests that prove my fix is effective or that my feature works.
~- [ ] I have added proper `data-cy` and `data-testid` attributes.~
- [x] I have added the necessary label (`patch`/`minor`/`major` - If package publish
      is required).

**Reviewers**

### CODE REVIEW

**Comment by farhanlatheef** [comment]
@josephmathew900 _a

**Review by josephmathew900 — COMMENTED** [review]
@farhanlatheef _a As discussed in the standup, I tested these changes locally in neetoPlanner and found that they are not working as expected. The API is getting triggered indefinitely.

image

**Comment by farhanlatheef** [comment]
@josephmathew900 _a

**Review by josephmathew900 — APPROVED** [review]
LGTM.

@farhanlatheef _a Please add the description also, it is required for [CHANGELOG.md](https://github.com/bigbinary/neeto-ui/blob/main/CHANGELOG.md). Please add the description and merge the PR.

File: src/components/DateTimePicker/index.jsx [hunk 1]
```diff
@@ -31,38 +37,54 @@ const DateTimePicker = ({
   datePickerProps,
   timePickerProps,
 }) => {
-  const [open, setOpen] = React.useState(false);
-  const [time, setTime] = useState(value);
-
+  const [open, setOpen] = useState(datePickerProps?.open);
+  const [date, setDate] = useState();
+  const [time, setTime] = useState();
+  const [changedField, setChangedField] = useState();
```
Comment (yedhink) [inline src/components/DateTimePicker/index.jsx:43]: Can we add an eslint rule to catch state declarations which doesn't have initial value? I think being explicit during state setting is better than assuming or reading the whole component's code to figure out what value a state should be dealing with. Here I think setting to `null` would be more explicit as it's an intentional empty value. 

Another example: The `changeField` state when I glanced through PR changes seems to be dealing with string values - if that's the case then we ought to set the initial value of this state with an empty string.

Refer:
- We had a section in React best practices book - that book itself is 404 right now. :/

What are your thoughts?

**Review by yedhink — COMMENTED** [review]
@josephmathew900 @farhanlatheef Noticed the following. Lemme know your thoughts.

File: src/components/DateTimePicker/index.jsx (hunk 1 above)
Comment (farhanlatheef) [reply src/components/DateTimePicker/index.jsx:43]: I can add explicit initial values. About the rule, well one of the reason people use Typescript.

Comment (josephmathew900) [reply src/components/DateTimePicker/index.jsx:43]: Yes we can add an eslint rule for this. I will create an issue for this.

Comment (josephmathew900) [reply src/components/DateTimePicker/index.jsx:43]: Issue: https://github.com/bigbinary/eslint-plugin-neeto/issues/561

**Pull request merged**